plugins/services/src/js/containers/service-connection/ServiceConnectionContainer.js

```diff
@@ -0,0 +1,189 @@
+  }
+
+  getPortDefinitionDetails(portDefinition) {
+    return (
+      <div>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            Protocol
+          </ConfigurationMapLabel>
+          {this.getProtocolValue(portDefinition)}
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            Container Port
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            {getDisplayValue(portDefinition.containerPort)}
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            Host Port
+          </ConfigurationMapLabel>
+          {this.getHostPortValue(portDefinition)}
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            Service Port
+          </ConfigurationMapLabel>
+          <ConfigurationMapValue>
+            {getDisplayValue(portDefinition.servicePort)}
+          </ConfigurationMapValue>
+        </ConfigurationMapRow>
+        <ConfigurationMapRow>
+          <ConfigurationMapLabel>
+            Load Balanced Address
+          </ConfigurationMapLabel>
+          {this.getLoadBalancedAddressValue(portDefinition)}
+        </ConfigurationMapRow>
+      </div>
+    );
+  }
+
+  getPortDefinitions(endpoints) {
+    return endpoints.map(portDefinition => {
+      return (
+        <ConfigurationMapSection>
+          <ConfigurationMapHeading>
+            {portDefinition.name}
+          </ConfigurationMapHeading>
+          {this.getPortDefinitionDetails(portDefinition)}
+        </ConfigurationMapSection>
+      );
+    });
+  }
+
+  getAlertPanelFooter() {
+    return (
+      <div className="button-collection flush-bottom">
+        <button
+          className="button"
+          onClick={this.handleOpenEditConfigurationModal}
+        >
+          Edit Configuration
+        </button>
+      </div>
+    );
+  }
+
+  render() {
+    const { service } = this.props;
+    let endpoints = [];
+    if (service.instances && service.instances.length > 0) {
+      service.instances.forEach(instance => {
+        instance.containers.forEach(container => {
+          endpoints = endpoints.concat(container.endpoints);
+        });
+      });
+    }
+
+    if (service.portDefinitions) {
+      endpoints = endpoints.concat(service.portDefinitions);
+    }
+
+    if (service.container && service.container.portMappings) {
+      endpoints = endpoints.concat(service.container.portMappings);
+    }
+
+    if (!endpoints || endpoints.length === 0) {
+      return (
+        <AlertPanel>
+          <AlertPanelHeader>No Endpoints</AlertPanelHeader>
+          <p className="tall">
+            There are no endpoints currently configured for
+            {" "}
+            {this.props.service.getId()}
+            .You can edit the configuration to add service endpoints.
```

Comment:
Please add the missing trailing space after the dot

Reply:
Added